tests/kafkatest/services/kafka/templates/kafka.properties

```diff
@@ -28,7 +28,7 @@ security.inter.broker.protocol={{ interbroker_listener.security_protocol }}
 {% endif %}
 
 {% for k, v in listener_security_config.client_listener_overrides.iteritems() %}
-{% if k.startswith('sasl.') %}
+{% if k.startswith('sasl.') and k != 'sasl.enabled.mechanisms' %}
```

Comment:
It may be better to check the list of configs that actually get prefixed with mechanism. I think only a small subset of SASL configs are actually prefixed with mechanism. The config docs for these indicate if they are mechanism-prefixed.

Reply:
docs say this:
prefixed with `listener.<name>.`:
- `connections.max.reauth.ms`
-  all `ssl...` configs
prefixed with `listener.<name>.<sasl-mechanism>.`:
- `connections.max.reauth.ms`
- `sasl.jaas.config`
- `sasl.login.callback.handler.class`
- `sasl.login.class`
- `sasl.server.callback.handler.class`

also from the docs it seems to imply that `sasl.enabled.mechanisms` should instead be at the global-level, not per listener... does that sound correct? @rajinisivaram

Reply:
I have tried running with the global `sasl.enabled.mechanisms` instead of per-listener and it seems to not work for me...

maybe we need to update the kafka docs?

Reply:
All listener configs can be prefixed with listener prefix (`listener.name.{name}.`). So docs probably dont explicitly say this. Only a few SASL configs can be prefixed with mechanism. I think the docs do explicitly state this and the five you listed above look correct. All listener-prefixed configs can also be at global level, but listener-prefixed ones have precedence. So you can't override listener configs with global configs.

Reply:
I've added a check that looks for each of these 5 configs specifically.